src/alternative-gas-token/L1WrappedTokenGateway.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity =0.8.24;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+
+import {IL1ERC20Gateway} from "../L1/gateways/IL1ERC20Gateway.sol";
+import {IWETH} from "../interfaces/IWETH.sol";
+
+contract L1WrappedTokenGateway {
+    using SafeERC20 for IERC20;
+
+    /*********
+     * Error *
+     *********/
+
+    /// @dev Thrown when someone try to send ETH to this contract.
+    error ErrorCallNotFromFeeRefund();
+
+    /*************
+     * Constants *
+     *************/
+
+    /// @dev The safe gas limit used to bridge WETH to L2.
+    uint256 private constant SAFE_GAS_LIMIT = 200000;
```

Comment:
May need to increase this value to `450000`. If the first WETH deposit goes through this contract, we need at least `400000` to successfully create WETH token in L2.

If the first WETH deposit goes through gateway router, it is ok to use `200000` in this contract.

Reply:
Yes, testing, and just realized it.